modules/llm-cache/ds/kv_state_cache_block.h

```diff
@@ -170,20 +167,21 @@ class KVStateCacheBlockBuilder : public ObjectBuilder {
 
   int16_t Split(KVStateCacheBlockBuilder* child, int index);
 
-  const std::shared_ptr<TensorBuilder<double>> GetKeyStateBuilder(int layer) {
+  const std::shared_ptr<TensorBuilder<uint8_t>> GetKeyStateBuilder(int layer) {
     return keyStateTensorBuilderList[layer];
   }
 
-  const std::shared_ptr<TensorBuilder<double>> GetValueStateBuilder(int layer) {
+  const std::shared_ptr<TensorBuilder<uint8_t>> GetValueStateBuilder(
+      int layer) {
     return valueStateTensorBuilderList[layer];
   }
 
-  const std::vector<std::shared_ptr<TensorBuilder<double>>>
+  const std::vector<std::shared_ptr<TensorBuilder<uint8_t>>>
```

Comment:
I believe return a `const std::vector<T> &` (i.e., const reference of the vector) would be better for these getters to avoid copy.

Reply:
It's a big change if we use the template to replace `TensorBuilder<uint8_t>`, could we open another issue to address it?

Reply:
Why a big change?

Reply:
It's under the basic class of llm kv cache, If we change the type, the loc will be large. If you don't think it's hard to review, we can implement it here.